pkg/tsdb/mysql/sqleng/sql_engine.go

```diff
@@ -642,3 +642,17 @@ func epochPrecisionToMS(value float64) float64 {
 
 	return value
 }
+
+func isProcessingDownstreamError(err error) bool {
+	downstreamErrors := []error{
+		data.ErrorInputFieldsWithoutRows,
+		data.ErrorSeriesUnsorted,
+		data.ErrorNullTimeValues,
+	}
+	for _, e := range downstreamErrors {
+		if errors.Is(err, e) {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
I see that this is duplicated 3 times. Do you think we can put this a shared place? Having it this way is totally ok but just want to check the possibility.

Reply:
Why don't we use `sqlds` in core data sources?

Reply:
Historical reasons. Similarly to why we have 2-3 sql query builders. We have tasks in backlog to unify the query editors.